### ISSUE TITLE
False positive on CA1001 / CA2213 after update

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#281]
After installing https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/1.0.1, the simplified repro below starts causing warnings. I believe these warnings are incorrect.

```csharp
using System;

public sealed class C : IDisposable
{
    private readonly N n;

    public C()
    {
        n = new N(this);
    }

    public void Dispose()
    {
    }

    private void Y()
    {
    }

    private sealed class N
    {
        private readonly C owner;

        public N(C owner)
        {
            this.owner = owner;
        }

        public void X()
        {
            owner.Y();
        }
    }
}
```

![fxcop-warn](https://cloud.githubusercontent.com/assets/10324372/9576188/ecf1a72c-4fd5-11e5-8edd-d43839bbd152.png)

**Comment by srivatsn** [comment]
We've decided to cut rule CA2213 given it's very high noise to signal ratio (https://github.com/dotnet/roslyn-analyzers/issues/695)
